### ISSUE TITLE
Kotlin 1.4 EAP SAMs

### ISSUE THREAD

**Issue by nwillc** [pinterest/ktlint#726]
## Expected Behavior

Ktlint should not reject SAMs in Kotlin 1.4:

`fun interface Extractor<T> {
    fun extract(rs: ResultSet): T
}`

## Observed Behavior
Using Kotlin 1.4 EAP, when KTlink encounters a valid  "SAM" (see above) it yields:

`Not a valid Kotlin file (expecting function name or receiver type) (cannot be auto-corrected)`

## Your Environment
* Version of ktlint used:
 Tool Version: 0.36.0

* Name and version (or code for custom task) of integration used (Gradle plugin, Maven plugin)
Gradle Plugin  "org.jlleitschuh.gradle.ktlint" "9.2.1" 

* Version of Gradle used (if applicable):
Gradle 6.3

* Operating System and version:
OS X

* Link to your project (if it's a public repository):
https://github.com/nwillc/funkjdbc/tree/feature/kotlin-1.4

**Comment by romtsn** [comment]
We are now publishing ktlint snapshots that are built with the latest Kotlin-preview version, please check how to try it out [here](https://github.com/pinterest/ktlint#kotlin-development-version-snapshot)

**Comment by byencho** [comment]
@romtsn I'm curious why this issue is closed because I'm still seeing it on the latest version (`0.41.0`).

**Comment by vitkhudenko** [comment]
@byencho Just checked on the latest (`0.41.0`) - the fix is there indeed.

**Comment by byencho** [comment]
So here's some sample code I'm checking against:

```kotlin
package com.example

fun interface KtLintTest {
    fun testMethod()
}
```

Using:
- ktlint: "0.36.0"
- "org.jlleitschuh.gradle:ktlint-gradle:9.2.1"

I get the following:

> Task :smartkit:ktlintMainSourceSetCheck
.../src/main/java/com/example/KtLintTest.kt:6:4: Not a valid Kotlin file (expecting function name or receiver type) (cannot be auto-corrected) ()

Which is the same as this original issue. When I upgrade to

- ktlint: "0.41.0"
- "org.jlleitschuh.gradle:ktlint-gradle:10.1.0"

I get this instead:

> A failure occurred while executing org.jlleitschuh.gradle.ktlint.worker.KtLintWorkAction
   > KtLint failed to parse file: .../src/main/java/com/example/KtLintTest.kt

If I make that a normal interface (instead of a functional one with "fun") then the issue goes away.

So I'm not sure what's going on. Maybe its a bad interaction with some other dependency, or something wrong with that particular plugin.

**Comment by vitkhudenko** [comment]
@byencho the issue you are experiencing is most likely caused not by the ktlint itself.

We use ktlint just by running the `sh ./scripts/ktlint.sh` command. So it looks like it is the `org.jlleitschuh.gradle:ktlint-gradle` that causes the issue for you.

**Comment by byencho** [comment]
@vitkhudenko Yeah it was unfortunately. There was a version hardcoded in the local configuration of the plugin for some reason. Updating that fixed the issue. Thanks for reminding me about this because now I've actually got it all working 👍
